nightshift/datastore_transactions.py

```diff
@@ -194,15 +322,57 @@ def retrieve_open_matched_resources_with_full_bib_obtained(
         session.query(Resource)
         .filter(
             Resource.libraryId == libraryId,
+            Resource.resourceCategoryId == resourceCategoryId,
             Resource.status == "open",
-            Resource.deleted == False,
             Resource.fullBib.isnot(None),
         )
         .all()
     )
     return resources
 
 
+def retrieve_processed_files(session: Session, libraryId: int) -> list[str]:
+    """
+    Retrieves file handles of all processed files for specific library.
+
+    Args:
+        session:                `sqlalchemy.Session` instance
+        libraryId:              `nightshift.datastore.Library.nid`
+
+    Returns:
+        list of file handles
+    """
+    instances = session.query(SourceFile).filter_by(libraryId=libraryId).all()
```

Comment:
This can be made a bit more efficient by doing `.query(SourceFile.handle) if that's the only field you need. It still returns a tuple so you still need the list comprehension, but saves having to fetch the whole object.

Reply:
Done.